integration-tests/testconfig/testconfig.go

```diff
@@ -0,0 +1,360 @@
+	Network               *ctf_config.NetworkConfig        `toml:"Network"`
+	Common                *Common                          `toml:"Common"`
+	OCR2                  *ocr2_config.Config              `toml:"OCR2"`
+	ConfigurationName     string                           `toml:"-"`
+}
+
+func (c *TestConfig) GetLoggingConfig() *ctf_config.LoggingConfig {
+	return c.Logging
+}
+
+func (c *TestConfig) GetPrivateEthereumNetworkConfig() *test_env.EthereumNetwork {
+	//TODO implement me
+	return nil
+}
+
+func (c *TestConfig) GetPyroscopeConfig() *ctf_config.PyroscopeConfig {
+	//TODO implement me
+	return nil
+}
+
+var embeddedConfigs embed.FS
+var areConfigsEmbedded bool
+
+func init() {
+	embeddedConfigs = embeddedConfigsFs
+}
+
+// Saves Test Config to a local file
+func (c *TestConfig) Save() (string, error) {
+	filePath := fmt.Sprintf("test_config-%s.toml", uuid.New())
+
+	content, err := toml.Marshal(*c)
+	if err != nil {
+		return "", errors.Wrapf(err, "error marshaling test config")
+	}
+
+	err = os.WriteFile(filePath, content, 0600)
+	if err != nil {
+		return "", errors.Wrapf(err, "error writing test config")
+	}
+
+	return filePath, nil
+}
+
+// MustCopy Returns a deep copy of the Test Config or panics on error
+func (c TestConfig) MustCopy() any {
+	return deepcopy.MustAnything(c).(TestConfig)
+}
+
+// MustCopy Returns a deep copy of struct passed to it and returns a typed copy (or panics on error)
+func MustCopy[T any](c T) T {
+	return deepcopy.MustAnything(c).(T)
+}
+
+func (c TestConfig) GetNetworkConfig() *ctf_config.NetworkConfig {
+	return c.Network
+}
+
+func (c TestConfig) GetChainlinkImageConfig() *ctf_config.ChainlinkImageConfig {
+	return c.ChainlinkImage
+}
+
+func (c TestConfig) GetCommonConfig() *Common {
+	return c.Common
+}
+
+func (c TestConfig) GetChainlinkUpgradeImageConfig() *ctf_config.ChainlinkImageConfig {
+	return c.ChainlinkUpgradeImage
+}
+
+func (c TestConfig) GetConfigurationName() string {
+	return c.ConfigurationName
+}
+
+func (c *TestConfig) AsBase64() (string, error) {
+	content, err := toml.Marshal(*c)
+	if err != nil {
+		return "", errors.Wrapf(err, "error marshaling test config")
+	}
+
+	return base64.StdEncoding.EncodeToString(content), nil
+}
+
+type Common struct {
+	Network   *string `toml:"network"`
+	InsideK8s *bool   `toml:"inside_k8"`
+	L2RPCUrl  *string `toml:"l2_rpc_url"`
+}
+
+func (c *Common) Validate() error {
+	if c.Network == nil {
+		return fmt.Errorf("network must be set")
+	}
+
+	switch *c.Network {
+	case "localnet":
+		break
+	case "testnet":
+		break
+	default:
+		return fmt.Errorf("network must be either 'localnet' or 'testnet'")
+	}
+
+	if c.InsideK8s == nil {
+		return fmt.Errorf("inside_k8 must be set")
+	}
+	if c.L2RPCUrl == nil && *c.Network == "testnet" {
+		return fmt.Errorf("l2_rpc_url must be set")
+	}
+
+	return nil
+}
+
+type Product string
+
+const (
+	OCR2 Product = "ocr2"
+)
+
+var TestTypesWithLoki = []string{"Soak", "Smoke"}
```

Comment:
pls delete, I'm removing it from `core` now.

Reply:
Removed